client/extensions/zoninator/state/data-layer/zones/index.js

```diff
@@ -24,8 +33,50 @@ export const requestZonesError = ( { dispatch }, { siteId } ) =>
 export const updateZonesList = ( { dispatch }, { siteId }, { data } ) =>
 	dispatch( updateZones( siteId, data ) );
 
+export const createZone = ( { dispatch }, action ) => {
+	const { data, form, siteId } = action;
+
+	dispatch( startSubmit( form ) );
+	dispatch( removeNotice( 'zoninator-zone-create' ) );
+	dispatch( http( {
+		method: 'POST',
+		path: `/jetpack-blogs/${ siteId }/rest-api/`,
+		query: {
+			body: JSON.stringify( data ),
+			json: true,
+			path: '/zoninator/v1/zones',
+		},
+	}, action ) );
+};
+
+export const announceZoneSaved = ( { dispatch }, { form, siteId }, next, { data } ) => {
+	dispatch( stopSubmit( form ) );
+	dispatch( updateZone( siteId, data ) );
+	dispatch( successNotice(
+		translate( 'Zone saved!' ),
+		{ id: 'zoninator-zone-create' },
+	) );
+};
+
+export const announceZoneCreated = ( { dispatch, getState }, action, next, response ) => {
+	const { siteId } = action;
+
+	page( `/extensions/zoninator/${ getSiteSlug( getState(), siteId ) }` );
+	announceZoneSaved( { dispatch }, action, next, response );
```

Comment:
Does `announceZoneSaved` need to be a separate function? Having a separate function is normally a good thing in and of itself, but I will admit that at first I was a bit confused as to how `announceZoneCreated` was different from `announceZoneSaved` and why 2 functions were used. The naming seems to be a bit of a misnomer as well since nothing is actually being announced in `announceZoneCreated`.

I would consider rolling the functionality from `announceZoneSaved` into `announceZoneCreated`. Or if you feel strongly for separate functions, then simplify the parameters that are passed to `announceZoneSaved`:

`announceZoneSaved( dispatch, action, response.data );`